### ISSUE TITLE
pandas vs. numpy for `y`

### ISSUE THREAD

**Issue by elcorto** [deeplearning540/lesson03#3]
```py
X = np.stack((df.bill_length_mm, df.flipper_length_mm), axis=-1)
y = df.species_.cat.codes
```

`X` is a numpy array but `y` is a pandas Series which creates problems down the line when people try to index it. Using `y = df.species_.cat.codes.values` returns an np array which fixes things (even though `y.value_counts()` is not available any longer.

**Comment by PrometheusPi** [comment]
Depending on the seed used in `train_test_split`, the index `0` is either included or not-included in `x0_test = X_test[0]`. For seeds as in the video of for `20220323`, index `0` is part of `y_test` thus leading to working code. If index `0` is  included in `y_train` accessing `y_test[0]` will resulting in a "KeyError".

**Comment by psteinb** [comment]
Thanks for looking into that, I added `y = df.species_.cat.codes.values` in the "solution" notebook with this commit https://github.com/deeplearning540/deeplearning540.github.io/commit/0d418b264b43758035a06600f8ae8fcb8ad8c475

The notebook in this repo doesn't contain any code that yields a similar problem. Thanks a lot for reporting.
